crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs

```diff
@@ -704,6 +705,16 @@ async fn update_single_agent_info(
 
     tracing::debug!(?agent_info_signed);
 
+    // Write to the local peer store. The agent info will also be stored via a local broadcast,
+    // except in the case of a zero arc where it gets filtered out, so we store it here too
+    // to be sure it makes it into the local store.
+    evt_sender
+        .put_agent_info_signed(PutAgentInfoSignedEvt {
```

Comment:
If this always runs then is there any chance of this put being different to the other? i.e. the data that somebody else thinks they've written gets overwritten with something subtly different.

Maybe this could be extracted to a function so that its at least called the same way and is easily traceable with 'find usages'?

Reply:
Good question, I will check on that. Yes, it's a bit unfortunate to need to do this, it could be cleaner.

As for pulling it into a function, isn't `put_agent_info_signed` already pretty concise function for storing agent info? What else could be pulled in?

Reply:
I suppose a `put_local_agent_info` method could be helpful to tie those to instances together

Reply:
> is there any chance of this put being different to the other?

Nope. For this particular data, this is irrelevant. The underlying database only cares about the most recently authored non-expired agent_info for each pub key. It doesn't matter if a different process triggered that agent_info to be generated.

Reply:
I think what @ThetaSinner is asking about is if the agent info could somehow change between this put and the one that happens via local broadcast. Which may not be the case now but could be in the future if we somehow change this later. So maybe it's a good idea to link the two a bit more, at least lexically/AST-wise so we have a better chance of understanding the that put is happening in two places at once, which is not ideal.

Reply:
I'm just saying the clobbering is irrelevant. But, all good : )

Reply:
I think it is if we somehow alter the info before the first put (like changing the arc) and then that change isn't reflected in the second put so it gets clobbered. It seems at least vaguely possible that that could happen.